core/src/test/java/org/javarosa/core/model/test/UploadExtensionTest.java

```diff
@@ -0,0 +1,76 @@
+package org.javarosa.core.model.test;
+
+import org.javarosa.core.model.QuestionDataExtension;
+import org.javarosa.core.model.QuestionDef;
+import org.javarosa.core.model.UploadQuestionExtension;
+import org.javarosa.core.test.FormParseInit;
+import org.javarosa.xform.parse.QuestionExtensionParser;
+import org.javarosa.xform.parse.UploadQuestionExtensionParser;
+import org.javarosa.xform.parse.XFormParseException;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+
+import java.util.Vector;
+
+/**
+ * Tests for UploadQuestionExtensionParser and UploadQuestionExtension
+ */
+public class UploadExtensionTest {
+
+    Vector<QuestionExtensionParser> extensionParsers;
+
+    @Rule
+    public ExpectedException exception = ExpectedException.none();
+
+    @Before
+    public void init() {
+        extensionParsers = new Vector<>();
+        extensionParsers.add(new UploadQuestionExtensionParser());
+    }
+
+    @Test
+    public void testParseMaxDimenWithPx() {
+        FormParseInit formWithPx = new FormParseInit("/test_upload_extension_1.xml", extensionParsers);
```

Comment:
Now that we're expanding a lot, can we add these to subfolders like

/xform_tests/test_upload...
?